dd-sdk-android/src/main/kotlin/com/datadog/android/Datadog.kt

```diff
@@ -21,54 +21,67 @@ import com.datadog.android.v2.api.context.UserInfo
 import com.datadog.android.v2.core.DatadogCore
 import com.datadog.android.v2.core.NoOpSdkCore
 import com.datadog.android.v2.core.internal.HashGenerator
+import com.datadog.android.v2.core.internal.SdkCoreRegistry
 import com.datadog.android.v2.core.internal.Sha256HashGenerator
 import java.util.concurrent.atomic.AtomicBoolean
 
 /**
  * This class initializes the Datadog SDK, and sets up communication with the server.
  */
+@Suppress("DEPRECATION") // TODO RUMM-3103 remove deprecated references
 @SuppressWarnings("TooManyFunctions")
 object Datadog {
 
-    // TODO RUMM-2984 Implement global registry
+    internal val registry = SdkCoreRegistry(internalLogger)
+
     /**
      * Temporary thing, until global registry is implemented.
      */
+    @Deprecated("Will be removed in RUMM-3103")
     var globalSdkCore: SdkCore = NoOpSdkCore()
         internal set
 
     internal var hashGenerator: HashGenerator = Sha256HashGenerator()
 
+    @Deprecated("Will be removed in RUMM-3103")
     internal val initialized = AtomicBoolean(false)
 
     // region Initialization
 
     /**
-     * Initializes the Datadog SDK.
+     * Initializes a named instance of the Datadog SDK.
+     * @param instanceName the name of the instance (or null to initialize the default instance
      * @param context your application context
      * @param credentials your organization credentials
      * @param configuration the configuration for the SDK library
-     * @param trackingConsent as the initial state of the tracking consent flag.
+     * @param trackingConsent as the initial state of the tracking consent flag
+     * @return the initialized SDK instance, or null if something prevents the SDK from
+     * being initialized
      * @see [Credentials]
      * @see [Configuration]
      * @see [TrackingConsent]
      * @throws IllegalArgumentException if the env name is using illegal characters and your
      * application is in debug mode otherwise returns false and stops initializing the SDK
```

Comment:
now `null` is returned

```suggestion
     * application is in debug mode otherwise returns null and stops initializing the SDK
```